src/lexer.coffee

```diff
@@ -116,10 +116,8 @@ exports.Lexer = class Lexer
     [..., prev] = @tokens
     forcedIdentifier = colon or prev? and
       (prev[0] in ['.', '?.', '::', '?::'] or
-      not prev.spaced and prev[0] is '@')
-
-    # IcedCoffeeScript addition
-    forcedIdentifier = false if id is 'defer'
+      # 'defer' here is IcedCoffeeScript addition
+      not prev.spaced and prev[0] is '@') and id isnt 'defer'
```

Comment:
I think the old code should say

` forcedIdentifier = false if id is 'defer' and not colon `

to be correct. `id is 'defer'` only "cancels" `forcedIdentifier` on the right side of the original `or`, it should be unaffected if it's `true` because of `colon`.

I'm not sure if what I'm saying here is true and I don't entirely understand the lexer module, so I just brought back the old (working) version.